File: php/src/Services/MerchantAssetsService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;

class MerchantAssetsService
{
    private Client $client;
    public function __construct(Client $client) { $this->client = $client; }
    public function getAppCoinAssetList(): array { return $this->client->post('/getAppCoinAssetList', []); }
    public function getAppCoinAsset(int $coinId): array { return $this->client->post('/getAppCoinAsset', ['coinId' => $coinId]); }
    public function getAppCollectFeeRecordList(array $data = []): array { return $this->client->post('/getAggregationFeeRecord', $data); }
}

File: ruby/lib/ccpayment/services/merchant_assets_service.rb
module CCPayment
  module Services
    class MerchantAssetsService
      def initialize(client); @client = client; end
      def get_app_coin_asset_list; @client.post('/getAppCoinAssetList', {}); end
      def get_app_coin_asset(coin_id); @client.post('/getAppCoinAsset', { coinId: coin_id }); end
      def get_app_collect_fee_record_list(data = {}); @client.post('/getAggregationFeeRecord', data); end
    end
  end
end
